src/panic.md

```diff
@@ -16,14 +16,42 @@ There are also language features that provide a level of control over panic beha
 > [!NOTE]
 > The standard library provides the capability to explicitly panic via the [`panic!` macro][panic!].
 
+r[panic.runtime]
+## Panic runtimes
+
+The actual behavior and implementation of a panic is controlled by the _panic runtime_. The panic runtime is a handler linked into the output which provides the necessary implementation for panicking.
+
+> [!NOTE]
+> The Rust standard library provides two panic runtimes: `panic_unwind` (which unwinds the stack and is potentially recoverable) and `panic_abort` (which aborts the process and is non-recoverable). The default runtime depends on the target platform, but is generally `panic_unwind` on platforms with native support for C++ exceptions.
+
+> [!NOTE]
+> The panic runtime can be chosen in `rustc` with the [`-C panic`] CLI flag when building any crate type except an rlib.
+
+> [!NOTE]
+> When compiling code that is guaranteed to be linked to a non-recoverable panic runtime, the optimizer may assume that unwinding across Rust frames is impossible, which can result in both code-size and runtime speed improvements.
```

Comment:
Isn't it more accurate to say that this is done based on the panic strategy? We do this within a crate when we see `-Cpanic=abort`, without knowing which runtime will be linked into the final atifact.